packages/esm-service-queues-app/src/patient-search/patient-search.workspace.tsx

```diff
@@ -1,40 +1,77 @@
-import React, { useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import { SearchTypes } from '../types';
 import PatientScheduledVisits from './patient-scheduled-visits.component';
 import VisitForm from './visit-form/visit-form.component';
 import {
+  ArrowLeftIcon,
   type DefaultWorkspaceProps,
-  ExtensionSlot,
-  usePatient,
-  useVisit,
+  displayName,
+  ErrorState,
+  PatientBannerContactDetails,
   PatientBannerPatientInfo,
-  PatientPhoto,
   PatientBannerToggleContactDetailsButton,
-  PatientBannerContactDetails,
-  displayName,
+  PatientPhoto,
+  usePatient,
+  useVisit,
 } from '@openmrs/esm-framework';
 import ExistingVisitFormComponent from './visit-form/existing-visit-form.component';
 import styles from './patient-search.scss';
+import { Button, DataTableSkeleton } from '@carbon/react';
+import { useScheduledVisits } from './hooks/useScheduledVisits';
+import isNil from 'lodash-es/isNil';
 
 interface PatientSearchProps extends DefaultWorkspaceProps {
+  handleBackToSearchList?: () => void;
   viewState: {
     selectedPatientUuid?: string;
   };
 }
 
-const PatientSearch: React.FC<PatientSearchProps> = ({ closeWorkspace, viewState }) => {
+const PatientSearch: React.FC<PatientSearchProps> = ({ closeWorkspace, viewState, handleBackToSearchList }) => {
+  const { t } = useTranslation();
   const { selectedPatientUuid } = viewState;
   const { patient } = usePatient(selectedPatientUuid);
   const { activeVisit } = useVisit(selectedPatientUuid);
   const [searchType, setSearchType] = useState<SearchTypes>(SearchTypes.SCHEDULED_VISITS);
-  const [newVisitMode, setNewVisitMode] = useState<boolean>(false);
   const [showContactDetails, setContactDetails] = useState(false);
+  const { appointments, isLoading, isError } = useScheduledVisits(selectedPatientUuid);
+  const hasAppointments = useMemo(
```

Comment:
This is probably fine, but I wouldn't use `useMemo` and `useCallback` if the operations are not expensive.

https://kentcdodds.com/blog/usememo-and-usecallback#why-is-usecallback-worse

Reply:
This is a problem that will be solved soon, hopefully, with the new [React compiler](https://react.dev/learn/react-compiler#what-does-the-compiler-do).